rosetta/retriever/retriever.go

```diff
@@ -29,17 +29,24 @@ import (
 	"github.com/optakt/flow-dps/rosetta/object"
 )
 
+type Config struct {
+	TransactionLimit int
+}
+
 type Retriever struct {
+	cfg Config
+
 	params    dps.Params
 	index     index.Reader
 	validate  Validator
 	generator Generator
 	invoke    Invoker
 }
 
-func New(params dps.Params, index index.Reader, validate Validator, generator Generator, invoke Invoker) *Retriever {
+func New(cfg Config, params dps.Params, index index.Reader, validate Validator, generator Generator, invoke Invoker) *Retriever {
```

Comment:
This signature would be:

```go
func New(params dps.Params, index index.Reader, validate Validator, generator Generator, invoke Invoker, options ...func(*Config))
```

And then it would start with:

```
  cfg := Config {
    TransactionLimit: 20,
    // define sensible defaults
  }

  for _, option := range options {
    option(&cfg)
  }
```

You can either put the config into the retriever, or just read the fields out, depending on what creates less overhead. If you want to open files based on config parameters, for example, it might not be worth keeping the config struct after using the file names.

And yes, this means we define two defaults; but they are quite distinct in their purpose. The default inside the package (here) is about a _sane_ limit that will never create issues and never be destructive. The default of the command line flag in the main package is a _sensible_ limit, in terms of what makes sense for the default use case of the binary.